table/tables/mutation_checker.go

```diff
@@ -138,9 +146,9 @@ func checkHandleConsistency(rowInsertion mutation, indexMutations []mutation, in
 		}
 		// NOTE: handle type can be different, see issue 29520
 		if indexHandle.IsInt() == insertionHandle.IsInt() && indexHandle.Compare(insertionHandle) != 0 {
-			return errors.Errorf("inconsistent handles in row and index insertions. index handle = %v, "+
-				"row handle = %v, index = %+v, row = %+v",
-				indexHandle, insertionHandle, m, rowInsertion)
+			err = ErrInconsistentHandle.GenWithStackByArgs(tableName, indexInfo.Name.O, indexHandle, insertionHandle, m, rowInsertion)
+			logutil.BgLogger().Error(err.Error())
```

Comment:
Consider:
```suggestion
			logutil.BgLogger().Error("inconsistent handles in row and index insertions detected", zap.Error(err))
```

Reply:
It would be better to use `zap.Error` to put the error to the log

Reply:
What's the benefit?

Reply:
It's the pattern that we always use. Usually the log message is a constant string and the other various information are provided in additional fields, wrapped with `zap` which makes it lazy-evaluated.